### DISCUSSION TITLE
Getting 404 but logs showing 200

### DISCUSSION THREAD

**Discussion by aaa3334** [vercel/next.js#66257] [App Router]
Hi!
I am at a loss of how to fix this so would love some help. I have been developing in localhost and just deployed to a website address on vercel yesterday.
Everything was working on localhost and most things work on vercel, except when middleware is redirecting me to a specific page (basically a signup route), I am getting a 404 error on the page... yet in the logs it shows as a 200 code and no errors.

This is an example of the route:
/dashboard/signupflow/step2

[Error] Failed to load resource: the server responded with a status of 404 () (step2, line 0)

If the user is redirected to this route (or anything with signupflow in it) it returns a 404... if not, it works...
I am sure I have a bug somewhere, my problem is it works 100% in localhost and I am getting a 200 in the logs on vercel so I am not sure where to look.

There is no layout file in any of these and some of the pages in this route are super basic.

Does anyone have any thoughts as to what I could investigate here? Or has had a similar issue before? 

Thanks in advance!

**Comment by aaa3334** [comment]
Still no clue what caused this... I changed the route from signupflow to signup_flow and it all suddenly worked....
If you encounter a similar issue my advice is to just change the wrapping folder name where the issue starts from. I am not sure why it happened and it is worrysome it was showing a 200 in the logs when the user receivs a 404.
I tried to do a reproduction but it al worked so I am not sure how to report it as a bug... (as it seems it was nothing wrong on my code end either).

**Comment by xentral-bnu** [comment]
Same in pages router when using `getStaticPaths` and `fallback: false` in production - the response is a 404 but it is being logged as a 200 in the vercel logs
